### PR TITLE
couple updates to best practices.

### PR DESCRIPTION
**Pull request by motobob**
couple readme updates as a follow up to #338

@kkuzmin more suggestions welcome

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -170,5 +170,10 @@ In general it follows default indentation rules of official erlang-mode as provi
 
 ## Best Practices ##
 
-- Public interfaces with paging logic should prefer {ok, Results, Marker} style to the {{paged, Marker}, Results} found in some modules
-- Public interfaces should normally expose proplists over records
+- All interfaces should provide a method for working with non-default config.
+- Public interfaces with paging logic should prefer `{ok, Results, Marker}` style to the `{{paged, Marker}, Results}` found in some modules.
+In case of records output, tokens should part of the record.
+- Passing next page `NextToken`, `NextMarker` is preferred with `Opts` rather than a a fun parameter like found in many modules.
```
Comment (kkuzmin) [inline README.md]: typo, `a a`

File: README.md [hunk 2]
```diff
@@ -170,5 +170,10 @@ In general it follows default indentation rules of official erlang-mode as provi
...
+- Public interfaces should normally expose proplists over records.
+- Exposed records are to be used only for complex outputs. Examples to follow: ddb2, ecs.
+- Library should not expose any long running processes
```
Comment (kkuzmin) [inline README.md]: Please add some example here.

File: README.md [hunk 3]
```diff
@@ -170,5 +170,10 @@ In general it follows default indentation rules of official erlang-mode as provi
 
 ## Best Practices ##
 
-- Public interfaces with paging logic should prefer {ok, Results, Marker} style to the {{paged, Marker}, Results} found in some modules
-- Public interfaces should normally expose proplists over records
+- All interfaces should provide a method for working with non-default config.
+- Public interfaces with paging logic should prefer `{ok, Results, Marker}` style to the `{{paged, Marker}, Results}` found in some modules.
+In case of records output, tokens should part of the record.
+- Passing next page `NextToken`, `NextMarker` is preferred with `Opts` rather than a a fun parameter like found in many modules.
+- Public interfaces should normally expose proplists over records.
```
Comment (kkuzmin) [inline README.md]: Do we still prefer proplists over records for simple outputs? What about records for all newly added modules?

**Comment by motobob** [comment]
addressed

**Comment by kkuzmin** [comment]
@motobob thank you 👍

**Pull request merged**